library/src/main/java/com/github/mproberts/rxdatabinding/bindings/RecyclerViewDataBindings.java

```diff
@@ -340,6 +344,28 @@ private RecyclerViewAdapter(ItemDataProvider dataProvider, ItemViewCreator viewC
         public void onAttachedToRecyclerView(RecyclerView recyclerView) {
             super.onAttachedToRecyclerView(recyclerView);
 
+            if (_lifecycle == null) {
+                _lifecycle = new WindowAttachLifecycle(recyclerView);
```

Comment:
Should you be creating a new window attach lifecycle regardless? The lifecycle should overwrite and replace the existing lifecycle if the recycler swaps

Reply:
Yeah I can do that. Technically, it won't really change anything though, will it?